internal/hardware/validator.go

```diff
@@ -197,6 +212,13 @@ func totalizeRequirements(ocpRequirements models.ClusterHostRequirementsDetails,
 				total.InstallationDiskSpeedThresholdMs = details.InstallationDiskSpeedThresholdMs
 			}
 		}
+		if details.NetworkLatencyThresholdMs > 0 && (total.NetworkLatencyThresholdMs == 0 || details.NetworkLatencyThresholdMs < total.NetworkLatencyThresholdMs) {
```

Comment:
You probably want `min` here. That is, from all the different type of requirements (ocp, operators, etc...), pick the lowest packet loss or network latency required to fulfill that group's needs so that the host requirements match it.

The caveat here is that the default value when none specified is 0, and 0 is a valid number for both requirements, so we have to be careful that we don't overwrite whatever value is greater than 0 when none other is defined.

I'll create a helper function to take care of this particular case.

Reply:
Ok, so after further thinking, I realized it was not going to work because `0` is both provided as default value and a value that can be set in the configuration. 

In the current shape it is not possible to discern which one is which. The only way is to define a null value for default/undefined. However, swagger 2.0 does not offer that possibility (swagger 3.0 provides a new field called `nullable`, but I did not have a chance to test it nor it would make any difference here since were not going to migrate to swagger 3.0 because of this case :wink: ).

So I overcame this by changing the swagger definition and wrapping the field with an object that contains a field `Value` of type `float64`. If no value is provided, the field is `null`.

This means changes to how the values are defined in json now will look like this:
```json
"network_latency_threshold_ms":{"value":100}
```
instead of how it was before:
```json
"network_latency_threshold_ms":100
```

Funny thing is that this change makes these 2 validations disabled by default if no values are set. It's a side effect of the change, not intended :smile:

Suggestions are welcome :smile:

Reply:
1. You see how complicated this condition you had to explain it with multiple sentences? :) Please make that helper function with appropriate documentation.
2. We do use `x-nullable` in our `swagger.yaml` which is used by the go-swagger implementation. It makes the field a pointer so it could have a null value.
for example: https://github.com/openshift/assisted-service/blob/55bc6bf876425b9a5f1680af4f37767e2c043692/swagger.yaml#L4181-L4184

Reply:
Sweet! thanks!